### PR TITLE
expression: add builtin cot, degrees

### PR DESCRIPTION
**Pull request by mtunique**
## What have you changed? (mandatory)

add builtin function cot, degrees

## What are the type of the changes? (mandatory)
- New feature #3275

## How has this PR been tested? (mandatory)

UT

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_math.rs [hunk 1]
```diff
@@ -166,6 +166,25 @@ impl ScalarFunc {
...
+                return Ok(Some(cot));
+            }
+        }
+        Err(Error::overflow("DOUBLE", &format!("cot({})", n)))
+    }
+
+    #[inline]
+    pub fn degrees(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
+        let n = try_opt!(self.children[0].eval_real(ctx, row));
+        Ok(Some(n * 180.0 / f64::consts::PI))
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/builtin_math.rs]: How about `n.to_degrees()`?

Comment (mtunique) [reply src/coprocessor/dag/expr/builtin_math.rs]: changed

**Comment by hicqu** [comment]
LGTM.

**Comment by huachaohuang** [comment]
@mtunique Please sign-off the DCO.

**Comment by mtunique** [comment]
@huachaohuang Maybe the last two commits made by you are not signed-off the DCO?

**Comment by mtunique** [comment]
@Connor1996 Master doesn't contain the changes. Why you close this PR.

**Comment by Connor1996** [comment]
@mtunique  Sorry🙏, when I fix the DCO issue, it makes some commit away. I am working on fix this.

**Comment by Connor1996** [comment]
@mtunique oops, when this PR closed, I have no permission to push to your branch. Can you fix it in your local branch and push to remote branch:
```
git reflog #find your origin commit hash
git reset --hard {your_hash}
git push -f 
```
Sorry to bother you

**Comment by Connor1996** [comment]
@mtunique if your didn't do `pull` in your local branch after my push, `git push -f` is just enough

File: src/coprocessor/dag/expr/scalar_function.rs [hunk 2]
```diff
@@ -220,6 +220,11 @@ impl ScalarFunc {
             | ScalarFuncSig::Inet6Aton
             | ScalarFuncSig::Inet6Ntoa
             | ScalarFuncSig::UnHex
+            | ScalarFuncSig::MD5,
```
[the remark this answers is not in the document]
Comment (mtunique) [reply src/coprocessor/dag/expr/scalar_function.rs]: solved

**Comment by Connor1996** [comment]
@mtunique still have this `Update scalar_function.rs` commit😂

**Comment by mtunique** [comment]
@Connor1996 I try to rebase this three commits to one.

**Comment by mtunique** [comment]
@Connor1996 Done.

**Pull request merged**